homework_4/src/main/java/com/epam/spring/controller/dto/UserDto.java

```diff
@@ -0,0 +1,50 @@
+package com.epam.spring.controller.dto;
+
+import com.epam.spring.controller.dto.group.OnCreate;
+import com.epam.spring.controller.dto.group.OnUpdate;
+import com.epam.spring.service.constraints.anotations.Phone;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.Builder;
+import lombok.Data;
+
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Null;
+
+@Data
+@Builder
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class UserDto {
+
+    @NotBlank(message = "'surname' shouldn't be empty", groups = OnCreate.class)
+    private String surname;
+
+    @NotBlank(message = "'name' shouldn't be empty", groups = OnCreate.class)
+    private String name;
+
+    @NotBlank(message = "'patronymic' shouldn't be empty", groups = OnCreate.class)
+    private String patronymic;
+
+    @Email
+    @NotBlank(message = "'email' shouldn't be empty", groups = OnCreate.class)
+    private String email;
+
+    @NotBlank(message = "'phone' shouldn't be empty", groups = OnCreate.class)
+    @Phone
```

Comment:
Hi. Usage **@Validated** annotation with configured groups to launch validation requires corresponding groups to be configured on the constraint annotation, otherwise it will not apply. Thanks